lib/services/local/resource_test.go

```diff
@@ -0,0 +1,183 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/backend/lite"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/suite"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/jonboulle/clockwork"
+	"gopkg.in/check.v1"
+)
+
+func TestResourceHelpers(t *testing.T) { check.TestingT(t) }
+
+type ResourceSuite struct {
+	bk backend.Backend
+}
+
+var _ = fmt.Printf
+var _ = check.Suite(&ResourceSuite{})
+
+func (r *ResourceSuite) SetUpSuite(c *check.C) {
+	utils.InitLoggerForTests(testing.Verbose())
+}
+
+func (r *ResourceSuite) SetUpTest(c *check.C) {
+	var err error
+
+	clock := clockwork.NewFakeClockAt(time.Now())
+
+	r.bk, err = lite.NewWithConfig(context.TODO(), lite.Config{
+		Path:             c.MkDir(),
+		PollStreamPeriod: 200 * time.Millisecond,
+		Clock:            clock,
+	})
+	c.Assert(err, check.IsNil)
+}
+
+func (r *ResourceSuite) TearDownTest(c *check.C) {
+	c.Assert(r.bk.Close(), check.IsNil)
+}
+
+func (r *ResourceSuite) dumpResources(c *check.C) []services.Resource {
+	startKey := []byte{0}
+	endKey := []byte{
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	}
+	result, err := r.bk.GetRange(context.TODO(), startKey, endKey, 0)
+	c.Assert(err, check.IsNil)
+	var resources []services.Resource
+	for _, item := range result.Items {
+		rsc, err := DeitemizeResource(item)
+		c.Assert(err, check.IsNil)
+		resources = append(resources, rsc)
+	}
+	return resources
+}
+
+func (r *ResourceSuite) runCreationChecks(c *check.C, resources ...services.Resource) {
+	err := CreateResources(r.bk, resources...)
+	c.Assert(err, check.IsNil)
+	dump := r.dumpResources(c)
+Outer:
+	for _, exp := range resources {
+		for _, got := range dump {
+			if got.GetKind() == exp.GetKind() && got.GetName() == exp.GetName() && got.Expiry() == exp.Expiry() {
+				continue Outer
+			}
+		}
+		c.Errorf("Missing expected resource kind=%s,name=%s,expiry=%v", exp.GetKind(), exp.GetName(), exp.Expiry().String())
+	}
+}
+
+func (r *ResourceSuite) TestUserResource(c *check.C) {
+	alice := newUser("alice", nil)
+	bob := newUser("bob", nil)
+	// Check basic dynamic item creation
+	r.runCreationChecks(c, alice, bob)
+	// Check that dynamically created item is compatible with service
+	s := NewIdentityService(r.bk)
+	b, err := s.GetUser("bob")
+	c.Assert(err, check.IsNil)
+	if !bob.Equals(b) {
```

Comment:
could be a one liner: `c.Assert(bob.Equals(b), Equals, true, Commentf("dynamically inserted user does not match)`